docs/next/guides/building-and-tooling/modules.md

```diff
@@ -0,0 +1,1077 @@
+---
+title: Modules
+metaTitle: Modules - Guide - Handsontable Documentation
+permalink: /next/modules
+canonicalUrl: /modules
+tags:
+  - tree shaking
+---
+
+# Modules
+
+Import just the modules that you actually use, to reduce Handsontable's bundle size.
+
+[[toc]]
+
+## Overview
+
+Handsontable is a comprehensive tool with a broad range of features. If you don't use all of them, you can pick just the parts that you need, and get rid of the rest (e.g., unnecessary translations). This approach reduces Handsontable's impact on the overall size of your app.
+
+To make this possible, Handsontable is divided into modules.
+
+### Use modules
+
+To get the most out of using Handsontable's modules:
+1. Import the [core modules](#core-modules).
+2. Import [optional modules](#optional-modules) of your choice.
+3. Remove redundant code (tree shaking).
+
+#### Use modules in frameworks
+
+To find out how to use modules with Handsontable's framework wrappers, see the following guides:
+- [Modules in React](@/guides/integrate-with-react/react-modules.md)
+- [Modules in Angular](@/guides/integrate-with-angular/angular-modules.md)
+- [Modules in Vue 2](@/guides/integrate-with-vue/vue-modules.md)
+- [Modules in Vue 3](@/guides/integrate-with-vue3/vue3-modules.md)
+
+### Bundler support
+
+To use Handsontable's modules, your environment needs to support the `import` syntax, which is typically provided by a bundler.
+
+We successfully tested Handsontable with the following bundlers:
+- webpack
+- Parcel
+- Rollup
+- Vite
+
+If Handsontable's modules don't work with your bundler, report it as a [bug](https://github.com/handsontable/handsontable/issues/new?assignees=&labels=&template=01-handsontable.md&title=).
+
+## Core modules
+
+No matter which of the optional modules you use, the following modules are always required:
+
+- Handsontable's core functionalities
+- The default cell type: [`text`](@/guides/cell-types/cell-type.md#available-cell-types)
+- Handsontable's CSS
+- Required dependencies:
+  - [moment.js](https://momentjs.com/) (you can easily [optimize](@/guides/optimization/bundle-size.md#optimize-moment-js) its size)
+  - [DOMpurify](https://www.npmjs.com/package/dompurify)
+  - [core-js](https://www.npmjs.com/package/core-js)
+
+### Import core modules
+
+To import the core modules:
```

Comment:
I am not sure how it is helpful that we refer to `handsontable/base` as "core module". The word "base" is right in the name, so I think we should refer to it as "base module" to avoid confusion.

Reply:
@warpech - done: [7fa2122](https://github.com/handsontable/handsontable/pull/9059/commits/7fa21224300a26779d485e3fd2835cd264cc54f8)